Lib/timeit.py

```diff
@@ -316,66 +373,12 @@ def main(args=None, *, _wrap_timer=None):
     if _wrap_timer is not None:
         timer = _wrap_timer(timer)
 
-    t = Timer(stmt, setup, timer)
-    if number == 0:
-        # determine number so that 0.2 <= total time < 2.0
-        callback = None
-        if verbose:
-            def callback(number, time_taken):
-                msg = "{num} loop{s} -> {secs:.{prec}g} secs"
-                plural = (number != 1)
-                print(msg.format(num=number, s='s' if plural else '',
-                                 secs=time_taken, prec=precision))
-        try:
-            number, _ = t.autorange(callback)
-        except:
-            t.print_exc()
-            return 1
-
-        if verbose:
-            print()
-
     try:
-        raw_timings = t.repeat(repeat, number)
-    except:
-        t.print_exc()
+        run(stmt, setup, timer, repeat, number, time_unit, bool(verbose))
+        return 0
+    except Exception:
+        timer.print_exc()
```

Comment:
`timer` is not the same as `t`. Can we move this inside `run()` and return the exit code?

Reply:
I was thinking about this, but it causes some issues:
- Why would this function return a number? It seems like an implementation detail leaking
- Why would we print the exception instead of throwing it if we're in a Python context?

Reply:
```suggestion
        t.print_exc()
```